docpages/example_programs/the_basics/using_callback_functions.md

```diff
@@ -0,0 +1,97 @@
+\page callback-functions Using Callback Functions
+
+When you create or get an object from Discord, you send the request to its API and in return you get either an error or the object you requested/created. Let's create a message and get it:
```

Comment:
Could we get a more in-depth explanation here?
Something like "you can pass a callable object like a function to API calls, your function will be executed when the request completes and it will be passed an object that has either an error or the object you requested/created"
Maybe something that mentions references and object lifetime and links to https://dpp.dev/lambdas-and-locals.html as well

Reply:
I will think about it, you can say I'm on it.

Reply:
I put some text there, hope it's good